### DISCUSSION TITLE
Dojo-Dungeon: Merge-Tage einführen?

### DISCUSSION THREAD

**Discussion by tgrothe** [Programmiermethoden/Dungeon#1465] [Ideas]
Das ist erst mal nur eine Idee.

Können wir für die Zeit des Dojo-Dungeon-Projekts Merge-Tage einführen?

- Dienstag EOD: Bis zu diesem Zeitpunkt sollten alle neuen Pull-Requests gestellt worden sein und sollten nicht mehr den Draft-Status haben.
- Mittwoch: Review-Tag. An diesem Tag dürfen keine Änderungen an den Pull-Requests mehr vorgenommen werden. Pull-Requests, die auf nicht auf Draft stehen, werden gereviewed. Das Team kann bereits mit dem folgenden Zyklus weitermachen.
- Donnerstag: Refactor-Tag. An diesem Tag werden Pull-Requests, die Änderungsvorschläge haben, refactored. Pull-Requests, die grün sind, können sofort gemerged werden.
- Freitag: Merge-Tag. An diesem Tag werden alle Pull-Requests, die grün sind, in struktureller Reihenfolge gemerged. Pull-Request, die Änderungsvorschläge haben, werden im folgenden Zyklus überarbeitet.

Vorteile:

- Feste Vorgabe und Struktur der Abläufe.
- Weniger Zeitaufwand für den Reviewer, weil nur noch an einem Tag ein Review/Approval erstellt werden muss.
- Weniger "Zwischenänderungen".

**Comment by cagix** [comment]
@tgrothe Es braucht keine Merge-Tage. Es braucht Beiträge, die mergebar sind!

Edit: Ich habe bisher nichts gesehen, wo ich drüber schauen möchte und ersthaft über einen Merge nachdenken würde. Die Beiträge von @Denniso3 und @JudiTeller sind ziemlich überschaubar bisher, und die von Dir sind einfach noch nicht reif zum Merge. Gib mir was mit der passenden Qualität und ich merge das jederzeit.

**Reply by tgrothe** [reply]
> @tgrothe Es braucht keine Merge-Tage. Es braucht Beiträge, die mergebar sind!

Ja, schon klar. Aber manchmal habe ich das Gefühl, deine Reviews sind etwas übergenau. Aber es kann auch an mir liegen.

**Reply by cagix** [reply]
Denk nochmal drüber nach.

**Reply by cagix** [reply]
@tgrothe Mein Verschlag ist, dass ihr euch irgendwo einen Fork anlegt. Dort könnt ihr nach Herzenslust rumschludern... und am Ende einen PR gegen das Hauptrepo einreichen. Die Frage ist nur, ob das dann in einem mergefähigen Zustand ist.

**Reply by tgrothe** [reply]
@cagix Ich kann nicht für die anderen entscheiden, aber vielleicht können wir das nachher besprechen.

... andererseits stehen die betreffenden PRs doch weiterhin auf draft, und ich hatte nicht zum review aufgerufen.

**Reply by cagix** [reply]
@tgrothe Ja, dann kommt doch mal aus der Tüte. Es liegt doch nur an euch. Schiebt doch endlich mal vernünftigen Content rüber, und hört auf die Review-Kommentare. Und schaut euch immer wieder auch das gesamte Bild an, nicht nur die Pinsel und/oder einzelne Farben! Ich habe auch keine Lust, meine Zeit mit diesem Ping-Pong auf Zweitsemestlerlevel zu vergeuden!

Edit: fix autocompletion nonsense
